### ISSUE TITLE
Feature request: DCE based on liveness analysis

### ISSUE THREAD

**Issue by playXE** [bnjbvr/regalloc.rs#27]
It should be possible to do dead code elimination based on liveness analysis information.

**Comment by bnjbvr** [comment]
Thanks for opening an issue! While you're totally right that we could, this is not part of the current goals of this crate. We expect the user of this crate to have done higher-level IR optimizations beforehand, including dead code elimination, etc. For instance, it is one thing done by Cranelift, our main target consumer. Regalloc can help just a bit by reducing the amount of dead spills/writes, but performing full DCE is out of scope. Does it make sense?

**Comment by playXE** [comment]
Yeah I understand. Maybe then it will be a good idea to provide liveness analysis API so everyone could use it for optimizations?

**Comment by Amanieu** [comment]
Alternatively, a way for the user to provide liveness information to regalloc so it doesn't have to be recomputed. But I think this is more of a long-term wishlist item, it is not needed for an MVP.
